testing/src/test/java/io/stargate/it/storage/ExternalStorage.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.it.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import com.datastax.oss.driver.api.core.Version;
+import com.datastax.oss.driver.api.testinfra.ccm.CcmBridge;
+import java.lang.reflect.AnnotatedElement;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** JUnit 5 extension for tests that need a backend database cluster managed by {@code ccm}. */
+public class ExternalStorage implements BeforeAllCallback, AfterAllCallback, ParameterResolver {
```

Comment:
In the future how hard will it be to include non-ccm managed storage?

Reply:
Just update this class to support it.

Reply:
We could use different `ClusterSpec` -like annotations too.

Reply:
Got it, thanks.